app/api/image_upload.py

```diff
@@ -25,4 +26,6 @@ def upload_image():
             image_file,
             UPLOAD_PATHS['temp']['image'].format(uuid=uuid.uuid4())
         )
+    if app.config['STATIC_DOMAIN']:
```

Comment:
Add this logic in the `upload_local` method itself. This should apply only for local uploads.